app/controllers/builds_controller.rb

```diff
@@ -1,6 +1,13 @@
 class BuildsController < ApplicationController
-  before_action :authorize_deployer!
-  before_action :find_project
+  include CurrentProject
+  include ProjectLevelAuthorization
+
+  before_action do
+    find_project(params[:project_id])
+  end
```

Comment:
I would move this fully into the CurrentProject module so it's not cut-pasted in all the controllers. Ruby modules have access to the class instance variables, so you can access the 'params' inside it.